### ISSUE TITLE
Multiple changes (EX via Ctrl+D) can lead to OOM due to multiple simultaneous linter runs

### ISSUE THREAD

**Issue by AccidentallyTheCable** [ansible/vscode-ansible#765]
### Summary

TLDR: When using a small system (EX 1GB ram, for a small VM), and performing mass replacements/insertions with Ctrl+D can cause OOM errors.

VM is a tiny VM specifically for only running ansible, it only has 1GB allocated. Using VSCode Remote via SSH, and working on the ansible playbooks. When I wish to do something to many things in the same file, such as fixing a bad quote from copy/pasta, this leads to many instances of ansible-lint running side by side, essentially triggered from each change. If enough changes are performed, this eventually causes an OOM event on the system, and VSCode has to go through the whole connection process all over again.

Replication:

1. Create a small VM (<2GB of RAM), with a decent sized playbook.
2. Connect remotely.
3. Use Ctrl+D (Add Selection to Next Find Match) to perform many insertions/replacements.
4. Monitor Memory, and `ps aux`

Suggested Resolution:

Refactor so that ansible-lint runs after X time (EX: 500ms, or make configurable) of idle after changes instead of on every change

### Extension version

1.1.34

### VS Code version

1.74.3

### Ansible Version

```shell
ansible [core 2.14.1]
  config file = /usr/local/share/ansible/ansible.cfg
  configured module search path = ['/usr/local/share/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.9/dist-packages/ansible
  ansible collection location = /usr/local/share/ansible/.ansible/collections
  executable location = /usr/local/bin/ansible
  python version = 3.9.2 (default, Feb 28 2021, 17:03:44) [GCC 10.2.1 20210110] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### OS / Environment

Debian Bullseye (Where Ansible is), Windows 10 (Where VSCode is)

**Comment by AccidentallyTheCable** [comment]
Caught PS Output during one of these incidents.
![Capture](https://user-images.githubusercontent.com/24231493/214493400-be5366e5-a856-4403-b8d8-bdb835733e5e.PNG)
